### PR TITLE
feat(logs): add distribution property to the Subscription class

### LINKED ISSUE #30422: logs: enable distribution setting on subscription filter
### Describe the feature

By setting [distribution property](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_logs.CfnSubscriptionFilter.html#distribution) in L1 Construct, You can choose the method used to distribute log data to the destination, which can be either random or grouped by log stream.

However, the L2 Construct does not support this feature.

This property is only applicable when the destination is an Amazon Kinesis data stream.

### Use Case

When you want to distribute log data randomly to Kinesis Data Stream.

### Proposed Solution

Add distribution property to the SubscriptionFilter class.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.143.0

### Environment details (OS name and version, etc.)

All

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #30422

### Reason for this change
Missing Property in the Subscription class.

### Description of changes
Add destination property to the Subscription class.

### Description of how you validated changes
Add unit tests and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-logs/lib/subscription-filter.ts [hunk 1]
```diff
@@ -57,6 +58,10 @@ export class SubscriptionFilter extends Resource {
       physicalName: props.filterName,
     });
 
+    if (props.distribution && !(props.destination instanceof KinesisDestination)) {
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-logs/lib/subscription-filter.ts:61]:
```suggestion
    if (
      props.distribution &&
      !Token.isUnresolved(props.distribution) && 
      !Token.isUnresolved(props.destination) && 
      !(props.destination instanceof KinesisDestination)
    ) {
```
We should check for unresolved values on validation.

File: packages/aws-cdk-lib/aws-logs/README.md [hunk 2]
```diff
@@ -120,6 +120,25 @@ new logs.SubscriptionFilter(this, 'Subscription', {
 });
 ```
 
+When you use `KinesisDestination`, you can choose the method used to
+distribute log data to the destination by setting the `Distribution` property.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-logs/README.md:124]:
```suggestion
distribute log data to the destination by setting the `distribution` property.
```

File: packages/aws-cdk-lib/aws-logs/README.md [hunk 3]
```diff
@@ -120,6 +120,25 @@ new logs.SubscriptionFilter(this, 'Subscription', {
...
+
+declare const stream: kinesis.Stream;
+declare const logGroup: logs.LogGroup;
+
+new logs.SubscriptionFilter(this, 'Subscription', {
+  logGroup,
+  destination: new destinations.KinesisDestination(stream),
+  filterPattern: logs.FilterPattern.allTerms("ERROR", "MainThread"),
+  filterName: 'ErrorInMainThread',
+  distribution: logs.Distribution.RANDOM, // distribution
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-logs/README.md:138]:
```suggestion
  distribution: logs.Distribution.RANDOM,
```

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍 Left some suggestions for minor adjustments

**Comment by mazyu36** [comment]
@lpizzinidev 
Thank you for your review! 
I have addressed your comments.

**Comment by GavinZZ** [comment]
Updated the title to correctly reflect the changes. This PR is adding `distribution` not `destination`.

**Pull request merged**